pkg/server/tls_windows.go

```diff
@@ -0,0 +1,342 @@
+package server
+
+import (
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"fmt"
+	"regexp"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/github/smimesign/certstore"
+)
+
+func (l *tlsListener) applyWindowsCertificateStore(c TLSConfig) error {
+
+	// Restrict normal TLS options when using windows certificate store
+	if c.TLSCertPath != "" {
+		return fmt.Errorf("cannot include certificate file when using windows certificate store")
+	}
+	if c.TLSKeyPath != "" {
+		return fmt.Errorf("cannot include key file when using windows certificate store")
+	}
+
+	var subjectRegEx *regexp.Regexp
+	var err error
+	if c.WindowsCertificateFilter.ClientSubjectRegEx != "" {
+		subjectRegEx, err = regexp.Compile(c.WindowsCertificateFilter.ClientSubjectRegEx)
+		if err != nil {
+			return fmt.Errorf("error compiling subject common name regular expression %w", err)
+		}
+	}
+
+	// If there is an existing windows certhandler notify it to stop refreshing
+	if l.windowsCertHandler != nil {
+		l.windowsCertHandler.stopUpdateTimer <- struct{}{}
+	}
```

Comment:
This will block forever if the handler goroutine exits or panics. I'd recommend passing a context to the cert handler and storing the cancel function, calling it if it's not nil:

```
if l.cancelCertHandler != nil {
   l.cancelCertHandler() 
   l.cancelCertHandler = nil 
}
```